lib/generate-doc.js

```diff
@@ -5,69 +5,108 @@ const { get: getSchema, set: setSchema } = require('./layer-schema')
+        // Add route params to schema
+        if (routeLayer && routeLayer.keys && routeLayer.keys.length) {
+          const keys = {}
 
-        const params = routeLayer.keys.map((k) => {
-          let param
-          if (schema.parameters) {
-            param = schema.parameters.find((p) => p.name === k.name && p.in === 'path')
-          }
-
-          // Reformat the path
-          keys[k.name] = '{' + k.name + '}'
+          const params = routeLayer.keys.map(k => {
+            let param
+            if (schema.parameters) {
+              param = schema.parameters.find(
+                p => p.name === k.name && p.in === 'path'
+              )
+            }
 
-          return Object.assign({
-            name: k.name,
-            in: 'path',
-            required: !k.optional,
-            schema: { type: 'string' }
-          }, param || {})
-        })
+            // Reformat the path
+            keys[k.name] = '{' + k.name + '}'
 
-        if (schema.parameters) {
-          schema.parameters.forEach((p) => {
-            if (!params.find((pp) => p.name === pp.name)) {
-              params.push(p)
-            }
+            return Object.assign(
+              {
+                name: k.name,
+                in: 'path',
+                required: !k.optional,
+                schema: { type: 'string' }
+              },
+              param || {}
+            )
           })
-        }
 
-        operation.parameters = params
-        path = pathToRegexp.compile(path)(keys, { encode: (value) => value })
-      }
+          if (schema.parameters) {
+            schema.parameters.forEach(p => {
+              if (!params.find(pp => p.name === pp.name)) {
+                params.push(p)
+              }
+            })
+          }
+
+          operation.parameters = params
+          path = pathToRegexp.compile(path)(keys, { encode: value => value })
+        }
 
-      doc.paths[path] = doc.paths[path] || {}
-      doc.paths[path][layer.method] = operation
-      setSchema(layer.handle, operation)
+        doc.paths[path] = doc.paths[path] || {}
+        doc.paths[path][layer.method] = operation
+        setSchema(layer.handle, operation)
+      })
     })
-  })
 
   return doc
 }
 
 function iterateStack (path, routeLayer, layer, cb) {
   cb(path, routeLayer, layer)
+
+  // handle nested routes
+  if (layer.name === 'router') {
+    layer.handle.stack.forEach(l => {
+      path = path || ''
+      iterateStack(path + split(layer.regexp).join('/'), layer, l, cb)
+    })
+  }
+
   if (!layer.route) {
     return
   }
-  layer.route.stack.forEach((l) => iterateStack(path + layer.route.path, layer, l, cb))
+  layer.route.stack.forEach(l =>
+    iterateStack(path + layer.route.path, layer, l, cb)
+  )
+}
+
+// https://github.com/expressjs/express/issues/3308#issuecomment-300957572
+function split (thing) {
+  if (typeof thing === 'string') {
+    return thing.split('/')
+  } else if (thing.fast_slash) {
```

Comment:
I believe this should be `if(typeof thing.fast_slash !== 'undefined'){`. In some cases fast_slash in regex path can exist but it can be set to false. In those cases, generate-doc blows up with this exception

``` 
 error TypeError: split(...).join is not a function
    at /var/www/**********/api/node_modules/@wesleytodd/openapi/lib/generate-doc.js:73:47
    at Array.forEach (<anonymous>)
    at iterateStack (/var/www/**********/api/node_modules/@wesleytodd/openapi/lib/generate-doc.js:71:24)
    at /var/www/**********/api/node_modules/@wesleytodd/openapi/lib/generate-doc.js:73:7
    at Array.forEach (<anonymous>)
    at iterateStack (/var/www/**********/api/node_modules/@wesleytodd/openapi/lib/generate-doc.js:71:24)
    at /var/www/**********/api/node_modules/@wesleytodd/openapi/lib/generate-doc.js:17:5
    at Array.forEach (<anonymous>)
    at generateDocument (/var/www/**********/api/node_modules/@wesleytodd/openapi/lib/generate-doc.js:16:26)
    at OpenApiMiddleware (/var/www/**********/api/node_modules/@wesleytodd/openapi/index.js:41:29)
    at Layer.handle [as handle_request] (/var/www/**********/api/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/var/www/**********/api/node_modules/express/lib/router/index.js:328:13)
    at /var/www/**********/api/node_modules/express/lib/router/index.js:286:9
    at Function.process_params (/var/www/**********/api/node_modules/express/lib/router/index.js:346:12)
    at next (/var/www/**********/api/node_modules/express/lib/router/index.js:280:10)
    at dispatch (/var/www/**********/api/node_modules/compose-middleware/src/index.ts:49:42)
    at next (/var/www/**********/api/node_modules/compose-middleware/src/index.ts:56:16)
    at /var/www/**********/api/src/middlewares/RequestContextMiddleware.js:17:13
    at dispatch (/var/www/**********/api/node_modules/compose-middleware/src/index.ts:70:21)
    at next (/var/www/**********/api/node_modules/compose-middleware/src/index.ts:56:16)
    at /var/www/**********/api/src/middlewares/RequestContextMiddleware.js:12:22
    at Namespace.run (/var/www/**********/api/node_modules/cls-hooked/context.js:97:5)
    at /var/www/**********/api/src/middlewares/RequestContextMiddleware.js:12:12
    at dispatch (/var/www/**********/api/node_modules/compose-middleware/src/index.ts:70:21)
    at middleware (/var/www/**********/api/node_modules/compose-middleware/src/index.ts:85:12)
    at /var/www/**********/api/node_modules/compose-middleware/src/index.ts:19:45
    at Layer.handle [as handle_request] (/var/www/**********/api/node_modules/express/lib/router/layer.js:95:5)
    at trim_prefix (/var/www/**********/api/node_modules/express/lib/router/index.js:328:13)

```